src/main/java/Problem2/LCA.java

```diff
@@ -2,10 +2,36 @@
 
 import Problem1.TreeNode;
 
+import java.util.ArrayList;
+import java.util.List;
+
 public class LCA {
     public static TreeNode<Integer> lowestCommonAncestor(TreeNode<Integer> root, TreeNode<Integer> p, TreeNode<Integer> q) {
         // homework
         // hint: pay attention to the assumptions of this problem.
-        return null;    // place holder
+
+        //if root is null
+        //this should not be the case assuming p and q can be found in the root tree
+        if(root == null)
+            return null;
+
+        //if root equals an input, return this root
+        if(root.val == p.val || root.val == q.val)
+            return root;
+
+        //recursion to check each node in the tree
+        TreeNode left = lowestCommonAncestor(root.left, p, q);
+        TreeNode right = lowestCommonAncestor(root.right, p, q);
+
+        //if left and right have a node, return this root
+        if(left != null && right != null)
+            return root;
+
+        //if only right has a node, return root of right
+        if(right != null)
+            return right;
+
+        //else, return root of left
```

Comment:
what about the case when both left and right are null?

Reply:
If both left and right are null, line 35 returns left. Recursion allows it to replace 'null' when it can by using the right node.

Reply:
I also forgot to enable actions on this homework so the check didn't go through. Should I close this pull request and reopen it?

Reply:
@m-squared47 all good now